src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/AbstractQuteTemplateLinkCollector.java

```diff
@@ -203,8 +208,12 @@ private VirtualFile getTemplateFile(Module project, String templateFilePathWitho
 
 	@Nullable
 	private VirtualFile getVirtualFile(Module project, String templateFilePathWithoutExtension, String suffix) {
+		VirtualFile @NotNull [] roots = ModuleRootManager.getInstance(project).getContentRoots();
```

Comment:
I think the main problem is that  `templateFilePathWithoutExtension` starts with `src/main/resources/templates` (ex : `src/main/resources/templates/page`).  templateFilePathWithoutExtension should not starts with  `src/main/resources` but just with `templates` (we will need to change that in the future too to support several known templates folder to scan like web for Quarkus Web Asset, but let's keep simple for the moment)

If templateFilePathWithoutExtension  doesn't contains src/main/resources, you can use the same logic than  getTemplateBaseDir(Module javaProject) by using:

`List<VirtualFile> resourcesDirs = ModuleRootManager.getInstance(javaProject).getSourceRoots(JavaResourceRootType.RESOURCE);

instead of using ModuleRootManager.getInstance(project).getContentRoots(); and perhaps havoind haing a mergeUri method.